### PR TITLE
[Sc][Research] Remove cdpId and triggerType from storage

### PR DESCRIPTION
**Pull request by halaprix**
[ Remove cdpId and triggerType from storage](https://app.shortcut.com/oazo-apps/story/5710/sc-reesearch-remove-cdpid-and-triggertype-from-triggers-data)

### CODE REVIEW

**Comment by halaprix** [comment]
tasks are down - the need reworks that are not required at this point - they are commented out

File: contracts/AutomationBot.sol [hunk 1]
```diff
@@ -310,139 +287,125 @@ contract AutomationBot {
...
-                triggerId,
-                AutomationBotStorage.TriggerRecord(0, 0, false)
+        require(command.isExecutionLegal(triggerData), "bot/trigger-execution-illegal");
+        IAdapter adapter = IAdapter(getAdapterAddress(1));
+        (bool status, ) = address(adapter).delegatecall(
+            abi.encodeWithSelector(
+                adapter.getCoverage.selector,
+                triggerData,
+                msg.sender,
+                0x0000000000000000000000000000000000000000,
```
Comment (adamskrodzki) [inline contracts/AutomationBot.sol:357]: pass `coverageToken` here

File: contracts/AutomationBot.sol [hunk 2]
```diff
@@ -310,139 +287,125 @@ contract AutomationBot {
...
-    }
-
     //works correctly in context of automationBot
     function execute(
         bytes calldata executionData,
-        uint256 cdpId,
         bytes calldata triggerData,
         address commandAddress,
         uint256 triggerId,
         uint256 daiCoverage
```
Comment (adamskrodzki) [inline contracts/AutomationBot.sol:345]: add parameter `coverageToken`

File: contracts/Emitter.sol [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
```
Comment (adamskrodzki) [inline contracts/Emitter.sol:3]: this cannot be used without full security, delete it

**Review by adamskrodzki — CHANGES_REQUESTED** [review]

**Comment by adamskrodzki** [comment]
resolve conflicts

File: contracts/AutomationBot.sol (hunk 1 above)
Comment (halaprix) [reply contracts/AutomationBot.sol:315]: ✅

File: contracts/AutomationBot.sol (hunk 2 above)
Comment (halaprix) [reply contracts/AutomationBot.sol:303]: ✅

File: contracts/adapters/MakerAdapter.sol [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        if (canCall(triggerData, target) && !allowance) {
+            manager.cdpAllow(cdpId, target, 0);
+            // emit ApprovalRevoked(cdpId, target);
+        }
+    }
+
+    function getCoverage(
+        bytes memory triggerData,
+        address receiver,
+        address,
```
[the remark this answers is not in the document]
Comment (halaprix) [reply contracts/adapters/MakerAdapter.sol:82]: ✅

File: contracts/AutomationBot.sol [hunk 5]
```diff
@@ -310,139 +287,125 @@ contract AutomationBot {
...
-
         address automationBot = serviceRegistry.getRegisteredService(AUTOMATION_BOT_KEY);
 
-        BotLike(automationBot).removeRecord(cdpId, triggerId);
+        BotLike(automationBot).removeRecord(triggerData, triggerId);
 
         if (removeAllowance) {
-            manager.cdpAllow(cdpId, automationBot, 0);
-            emit ApprovalRemoved(cdpId, automationBot);
+            IAdapter adapter = IAdapter(getAdapterAddress(1));
```
[the remark this answers is not in the document]
Comment (halaprix) [reply contracts/AutomationBot.sol]: it was a duplicate - `permit` removed from `removeTrigger`

**Review by adamskrodzki — APPROVED** [review]

**Review by adamskrodzki — CHANGES_REQUESTED** [review]
resolve conflicts

**Pull request merged**